core/pin/pin.go

```diff
@@ -170,6 +158,18 @@ func (p *Pin) Listen(ctx context.Context, dbURL string) {
 					return
 				}
 
+				if p == nil {
+					s.mu.Lock()
+					var ok bool
+					p, ok = s.pins[pinName]
+					if !ok {
+						p = newPin(s.db, pinName, height)
```

Comment:
Should newPin be called insertPin and also do the map assignment and broadcast? Dunno.